src/core/Microsoft.PowerToys.Settings.UI/ViewModels/KeyboardManagerViewModel.cs

```diff
@@ -36,15 +72,26 @@ private async void OnEditShortcut()
         private async Task OnRemapKeyboardBackground()
         {
             Helper.AllowRunnerToForeground();
-            ShellPage.DefaultSndMSGCallback(Helper.GetSerializedCustomAction("Keyboard Manager", "RemapKeyboard", "Create Remap Keyboard Window"));
+            ShellPage.DefaultSndMSGCallback(Helper.GetSerializedCustomAction(PowerToyName, RemapKeyboardActionName, RemapKeyboardActionValue));
             await Task.CompletedTask;
         }
 
         private async Task OnEditShortcutBackground()
         {
             Helper.AllowRunnerToForeground();
-            ShellPage.DefaultSndMSGCallback(Helper.GetSerializedCustomAction("Keyboard Manager", "EditShortcut", "Create Edit Shortcut Window"));
+            ShellPage.DefaultSndMSGCallback(Helper.GetSerializedCustomAction(PowerToyName, EditShortcutActionName, EditShortcutActionValue));
             await Task.CompletedTask;
         }
+
+        private void OnConfigFileUpdate()
+        {
+            // Todo: update the UI data source here.

```

Comment:
Yes that would be helpful to use a named mutex here for the Json file. In that we can also get rid of the dummy file that i'm using to notify the completion of update operation but we still need some sort of mechanism to suppress multiple events fired from single update operation since there is no parameter in event to represent that it's fired on the completion of write operation.

Reply:
I have updated the implementation to use named mutex for synchronization of json file and currently I have removed the logic for grouping the events added a comment. We can address it separately at the time of linking the data to UI.